Src/WindowsServer/WindowsServer.Shared/AppServicesHeartbeatTelemetryModule.cs

```diff
@@ -0,0 +1,181 @@
+        }
+
+        /// <summary>
+        /// Gets a value to provide internal access to the only instance of this class that *should* be available.
+        /// </summary>
+        public static AppServicesHeartbeatTelemetryModule Instance
+        {
+            get => AppServicesHeartbeatTelemetryModule.instance;
+            private set => AppServicesHeartbeatTelemetryModule.instance = value;
+        }
+
+        /// <summary>
+        /// Initialize the default heartbeat provider for Azure App Services. This module
+        /// looks for specific environment variables and sets them into the heartbeat 
+        /// properties for Application Insights, if they exist.
+        /// </summary>
+        /// <param name="configuration">Unused parameter.</param>
+        public void Initialize(TelemetryConfiguration configuration)
+        {
+            this.isInitialized = this.UpdateHeartbeatWithAppServiceEnvVarValues();
+        }
+
+        /// <summary>
+        /// Ensure we've cleaned up our static Instance.
+        /// </summary>
+        public void Dispose()
+        {
+            // ensure our Instance variable is not kept around.
+            AppServicesHeartbeatTelemetryModule.instance = null;
+        }
+
+        /// <summary>
+        /// Signal the AppServicesHeartbeatTelemetryModule to update the values of the 
+        /// Environment variables we use in our heartbeat payload.
+        /// </summary>
+        /// <returns>A value indicating whether or not an update to the environment variables occurred.</returns>
+        public bool UpdateHeartbeatWithAppServiceEnvVarValues()
+        {
+            bool hasBeenUpdated = false;
+
+            try
+            {
+                var hbeatManager = this.GetHeartbeatPropertyManager();
+                if (hbeatManager != null)
+                {
+                    hasBeenUpdated = this.AddAppServiceEnvironmentVariablesToHeartbeat(hbeatManager, isUpdateOperation: this.isInitialized);
+                }
+            }
+            catch (Exception appSrvEnvVarHbeatFailure)
+            {
+                WindowsServerEventSource.Log.AppServiceHeartbeatPropertySettingFails(appSrvEnvVarHbeatFailure.ToInvariantString());
+            }
+
+            return hasBeenUpdated;
+        }
+
+        internal bool AddAppServiceEnvironmentVariablesToHeartbeat(IHeartbeatPropertyManager hbeatManager, bool isUpdateOperation = false)
+        {
+            bool hasBeenUpdated = false;
+
+            if (hbeatManager == null)
+            {
+                WindowsServerEventSource.Log.AppServiceHeartbeatSetCalledWithNullManager();
+            }
+            else
+            {
+                foreach (var kvp in WebHeartbeatPropertyNameEnvVarMap)
+                {
+                    try
+                    {
+                        // get the variable, then expand it (otherwise we get the name we queried for in the value)
```

Comment:
I am not sure i understood this. Why not just read ENV variable like done everywhere else? 
https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/develop/Src/WindowsServer/WindowsServer.Shared/AzureWebAppRoleEnvironmentTelemetryInitializer.cs#L65

Reply:
Agreed. There doesn't ever seem to be any need to handle environment variable expansion for this value. PR #870 does not attempt expansion.

Reply:
`ExpandEnvironmentVariable` will make this method slower, let's skip if unnecessary.